src/tools/recvfuzz/recvfuzz.cpp

```diff
@@ -184,13 +192,18 @@ UdpRecvCallback(
                 memcpy((void *)PacketCopy->DestCid, Packet.DestCid, Packet.DestCidLen);
                 PacketCopy->SourceCid = PacketCopy->DestCid + Packet.DestCidLen;
                 memcpy((void *)PacketCopy->SourceCid, Packet.SourceCid, Packet.SourceCidLen);
-                PacketQueue.push_back(PacketCopy);
+                PacketCopy->_.Next = nullptr;
+
+                PacketQueueLock.Acquire();
+                *PacketQueueTail = PacketCopy;
+                PacketQueueTail = (QUIC_RX_PACKET**)&PacketCopy->_.Next;
```

Comment:
IMO, since this is C++ code, we should just embrace it and simplify our life:
- Put the packet directly in the smart wrapper for memory management so we don't have to care about that part anymore
- Put the smart pointer in a std::dequeue to have a simple push / pop semantic instead of manipulating a C-style linked list by hand
- ... and if we want to go the extra length, it then becomes very simple to wrap that in a struct that take the lock to make the push / pop operations thread safe

Reply:
At least for now, I prefer to use the C type which already has this Next pointer.